lib/facter/conjur.rb

```diff
@@ -11,8 +18,23 @@ def config
       end
     end
 
+    def config_registry
+      keyname = 'Software\\CyberArk\\Conjur'
+
+      values = []
+      result = Win32::Registry::HKEY_LOCAL_MACHINE.open(keyname) do |reg|
+        reg.each_value { |name, type, data| values << [name.gsub(/(.)([A-Z])/, '\1_\2').downcase, data] }
```

Comment:
I feel this would be clearer with a map. BTW, not sure if it's worth the trouble trying to comply with the camel case convention.

Reply:
You mean a map of the `'CamelCase' => 'under_score'` for each value?

Reply:
I actually meant something like `result = reg.map { |n, t, d| [name.gsub(...), data] }.to_h` :)

Reply:
Gotcha, ok. That's what I first though, though `reg` isn't enumerable on its own, so I wasn't sure.

I'll add another method in the mix to enumerate the values so that it can use `map` here to make it more clear. Thanks!

Reply:
Ah; but `each_value` is enumerable, so you can do `reg.each_value.map`.

Reply:
You are correct. I must be misremembering what wasn't working before. In any case, that cleans up nicely! 😄